go.work

```diff
@@ -0,0 +1,6 @@
+go 1.18
```

Comment:
Why do you need a new workspace ?

Reply:
It's used to allow `gopls` to work properly with disjoint Go modules in the same repo.

Reply:
I think this deserves a separate PR, where we should add the rest of Go modules to `go.work`